File: 1. Beginner/2712/2712.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        String[] days = {"MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY"};
        
        int n = scanner.nextInt();
        scanner.nextLine();
        
        for (int i = 0; i < n; i++) {
            String p = scanner.nextLine().trim();
            
            if (ver(p)) {
                char l = p.charAt(7);
                int d;
                
                if (l == '1' || l == '2') {
                    d = 0;
                } else if (l == '3' || l == '4') {
                    d = 1;
                } else if (l == '5' || l == '6') {
                    d = 2;
                } else if (l == '7' || l == '8') {
                    d = 3;
                } else {
                    d = 4;
                }
                
                System.out.println(days[d]);
            } else {
                System.out.println("FAILURE");
            }
        }
        
        scanner.close();
    }
    
    public static boolean ver(String p) {
        if (p.length() != 8) {
            return false;
        }
        
        if (p.charAt(3) != '-') {
            return false;
        }
        
        int k = 0;
        for (int i = 0; i < 3; i++) {
            char c = p.charAt(i);
            if (Character.isLetter(c) && Character.isUpperCase(c)) {
                k++;
            }
        }
        
        if (k != 3) {
            return false;
        }
        
        for (int i = 4; i < 8; i++) {
            char c = p.charAt(i);
            if (!Character.isDigit(c)) {
                return false;
            }
        }
        
        return true;
    }
}

File: 2712/2712.py
def ver(p):
    if len(p) != 8:
        return False

    if p[3] != '-':
        return False

    contador = sum(1 for c in p[:3] if c.isalpha() and c.isupper())
    if contador != 3:
        return False

    for c in p[4:]:
        if not c.isdigit():
            return False

    return True


days = ["MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY"]
for _ in range(int(input())):
    p = input().strip()

    if ver(p):
        l = p[-1]
        if l in ['1', '2']:
            d = 0
        elif l in ['3', '4']:
            d = 1
        elif l in ['5', '6']:
            d = 2
        elif l in ['7', '8']:
            d = 3
        elif l in ['9', '0']:
            d = 4
        print(days[d])
    else:
        print("FAILURE")